dao/authentication_dao.go

```diff
@@ -45,6 +50,10 @@ func init() {
 	GetAuthenticationDao = getDefaultAuthenticationDao
 }
 
+type AuthenticationDaoParams struct {
+	TenantID *int64
+}
```

Comment:
(non-blocking)

I like the idea of this for now - but I think we can take it a step even further and use the same params struct for all of the DAOs, but also include all of the request info in the struct. 

Having "only" the tenant_id was all we needed back in the day but now the list is:
- tenant_id + account_number + org_id
- user_id + maybe user_id
- (future) groups the user belongs to

Having all of these contained in a struct on the echo context that we can jsut pass aroudn will be nice. It also might be time for us to extend the `echo.Context` interface with our own methods. 

---

This will be something done in the future though - I'm ok with this implementation.